packages/server/src/unstable-core-do-not-import/createProxy.ts

```diff
@@ -1,26 +1,34 @@
 interface ProxyCallbackOptions {
-  path: string[];
-  args: unknown[];
+  path: readonly string[];
+  args: readonly unknown[];
 }
 type ProxyCallback = (opts: ProxyCallbackOptions) => unknown;
 
 const noop = () => {
   // noop
 };
 
+const freezeIfAvailable = (obj: object) => {
+  if (Object.freeze) {
```

Comment:
typeof checks are usually better for checking if globals exist but in this case Object is always defined so this should work?